### PR TITLE
Fix /std:c++17 on Windows + silent some warnings

### PR DESCRIPTION
**Pull request by bellenot**
- Fix compilation errors with `/std:c++17` (and later) option on Windows
- Add `/std:c++17` flag for ACLiC
- Add the `-wd4146` compiler flag to ignore the warning C4146: unary minus operator applied to unsigned type, result still unsigned (in LLVM)
- Add a the missing `#include <typeinfo>` and `#include <stdexcept>` with `/std:c++17`
- Don't add the `-fPIC` flag with MSVC (prevent the command line warning D9002: ignoring unknown option '-fPIC')

### CODE REVIEW

File: cmake/modules/SetUpWindows.cmake [hunk 1]
```diff
@@ -29,15 +29,15 @@ elseif(MSVC)
   math(EXPR VC_MINOR "${MSVC_VERSION} % 100")
 
   if(winrtdebug)
-    set(BLDCXXFLAGS "-Zc:__cplusplus -MDd -GR")
+    set(BLDCXXFLAGS "-Zc:__cplusplus -std:c++${CMAKE_CXX_STANDARD} -MDd -GR")
```
Comment (Axel-Naumann) [inline cmake/modules/SetUpWindows.cmake:32]:
```suggestion
    set(BLDCXXFLAGS "-std:c++${CMAKE_CXX_STANDARD} -MDd -GR")
```
`__cplusplus` doesn't seem to be needed anymore these days? https://www.godbolt.org/z/Kjfb1Tfa5

File: interpreter/CMakeLists.txt [hunk 2]
```diff
@@ -474,7 +474,7 @@ if (builtin_cling)
   set(CLING_CXXFLAGS ${CLING_CXXFLAGS} PARENT_SCOPE)
 
   string(APPEND CMAKE_CXX_FLAGS ${CLING_CXXFLAGS})
-  if (LLVM_ENABLE_PIC)
+  if (LLVM_ENABLE_PIC AND NOT MSVC)
```
Comment (Axel-Naumann) [inline interpreter/CMakeLists.txt:477]: Ouch thanks!

**Review by Axel-Naumann — APPROVED** [review]

File: cmake/modules/SetUpWindows.cmake (hunk 1 above)
Comment (bellenot) [reply cmake/modules/SetUpWindows.cmake:32]: It is, otherwise it is always 1997:  `int i = 199711L;`

Comment (Axel-Naumann) [reply cmake/modules/SetUpWindows.cmake:32]: Argh sorry misunderstood!

**Comment by bellenot** [comment]
The failure is unrelated. Merging

**Pull request merged**